### ISSUE TITLE
Fee returning code still not right!

### ISSUE THREAD

**Issue by n1bor** [ACINQ/eclair#596]
Have a real world example where we are out by 1 satoshi. Don't you hate floating point numbers!
Fix is to use Math.round instead of toLong

Can reproduce in REPL:
```
Welcome to Scala version 2.11.6 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_162).
Type in expressions to have them evaluated.
Type :help for more information.

scala> val fee=0.00003076
fee: Double = 3.076E-5

scala> (fee * 1e8).toLong
res0: Long = 3075

scala> fee * 1e8
res1: Double = 3075.9999999999995

scala> Math.round(fee * 1e8)
res2: Long = 3076

scala> 


```

**Comment by sstone** [comment]
Can you be a bit more specific about what you think is right/wrong with eclair's code ?

**Comment by n1bor** [comment]
Sorry - thought is was obvious. The last commit https://github.com/ACINQ/eclair/pull/589 is still not correct.
It receives a string from bitcoind over rpc with the fees that were in the transaction. But ".toLong" incorrectly converts that double returned by bitcoind into a long of satoshis.
Just because it rounds down a string of "0.00003076" when multiplied by 1e8 is a double of 3075.999999995 which gets rounded down to 3075 instead of 3076.

**Comment by rorp** [comment]
Math.round won't help in other cases. The ultimate fix is to avoid floating point numbers altogether.

**Comment by sstone** [comment]
Fixed by #616
